### ISSUE TITLE
Enable EM to use on-premises

### ISSUE THREAD

**Issue by swilliams11** [swilliams11/edgemicro-decorator#3]
The current implementation only allows public cloud.  A new feature should be added to allow EM to connect to on-premises installations. see below for example.  If the virtual host is not provided then the default values of default, secure will be used.  The virtual host should be a comma separate list of virtual hosts.  

Add a new Object to the 
{
    "credentials": {
    **
    "onpremises" : "true",
    "onprem_config": {
    "runtime_url": "runtime",
    "mgmt_url" : "mgmt",
    "virtual_host" : "default,secure"**
     }
     "application_name": "edgemicro_service",
     "edgemicro_port": "8080",
     "edgemicro_version": "2.3.1",
     "env": "test",
     "org": "org",
     "pass": "org",
     "tags": [
      "edgemicro"
     ],
     "user": "email@email.com"
    },
    "label": "user-provided",
    "name": "edgemicro_service",
    "syslog_drain_url": "",
    "tags": [],
    "volume_mounts": []
   }

**Comment by swilliams11** [comment]
Currently attempting to test this feature with an on-premises installation of Edge.

**Comment by swilliams11** [comment]
Strange issue with testing this with on-premise deployments.  I get the following error when I CF attempts to connect.

{ Error: read ECONNRESET
    at exports._errnoException (util.js:1022:11)
    at TCP.onread (net.js:569:26) code: 'ECONNRESET', errno: 'ECONNRESET', syscall: 'read' }

I tested it with AWS and GCP and it still generates this error.  I also updated the Application Security Group to include the IP address of the AWS and GCP and I still get the same error.  The same private cloud configuration works successfully from my local machine. So something in CF is  preventing the initial configure request when EM starts up.
